lib/ansible/modules/cloud/amazon/aws_secret.py

```diff
@@ -0,0 +1,404 @@
+        module.fail_json_aws(e, msg="Failed to delete secret")
+
+    return {
+        'secret_arn': '',
+        'version_id': '',
+        'changed': True
+    }
+
+
+def rotation_updater(client, module, secret_status):
+    if module.params.get('rotation_lambda') and 'RotationLambdaARN' not in secret_status['secret_details']:
+        rs_params = {}
+        rs_params['SecretId'] = secret_status['secret_arn']
+        rs_params['RotationLambdaARN'] = module.params.get('rotation_lambda')
+        rs_params['RotationRules'] = {
+            'AutomaticallyAfterDays': module.params.get('rotation_interval')
+        }
+        if module.params.get('client_request_token'):
+            rs_params['ClientRequestToken'] = module.params.get('client_request_token')
+
+        try:
+            response = client.rotate_secret(**rs_params)
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e, msg="Failed to add rotation policy to secret")
+        return {'changed': True, 'version_id': response['VersionId']}
+
+    if module.params.get('rotation_lambda') and 'RotationLambdaARN' in secret_status['secret_details']:
+        changes_detected = []
+        if secret_status['secret_details'].get('RotationLambdaARN') != module.params.get('rotation_lambda'):
+            changes_detected.append(True)
+        if secret_status['secret_details'].get('RotationRules').get('AutomaticallyAfterDays') != module.params.get('rotation_interval'):
+            changes_detected.append(True)
+
+        if any(changes_detected):
+            rs_params = {}
+            rs_params['SecretId'] = secret_status['secret_arn']
+            rs_params['RotationLambdaARN'] = module.params.get('rotation_lambda')
+            rs_params['RotationRules'] = {
+                'AutomaticallyAfterDays': module.params.get('rotation_interval')
+            }
+            if module.params.get('client_request_token'):
+                rs_params['ClientRequestToken'] = module.params.get('client_request_token')
+
+            try:
+                response = client.rotate_secret(**rs_params)
+            except (BotoCoreError, ClientError) as e:
+                module.fail_json_aws(e, msg="Failed to add rotation policy to secret")
+            return {'changed': True, 'version_id': response['VersionId']}
+
+        return {'changed': False}
+
+    if not module.params.get('rotation_lambda') and 'RotationLambdaARN' in secret_status['secret_details']:
+        try:
+            response = client.cancel_rotate_secret(
+                SecretId=secret_status['secret_arn']
+            )
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e, msg="Failed to remove rotation policy from secret")
+        return {'changed': True, 'version_id': ''}
+
+    return {'changed': False}
+
+
+def tag_updater(client, module, secret_status):
+    if module.params.get('tags') and 'Tags' not in secret_status['secret_details']:
```

Comment:
You can simplify this function by using compare_aws_tags. And it would be good to add a purge_tags option in line with other AWS modules so a user can choose whether or not tags omitted from their task should be removed or left alone.

To be consistent with other AWS modules, it will look something like:
```
current_tags = boto3_tag_list_to_ansible_dict(secret_status['secret_details'].get('Tags', []))
if module.params['tags'] is not None:  # user should be able to specify tags as an empty dict {} to remove all tags if they want
    tags_to_add, tags_to_remove = compare_aws_tags(current_tags, module.params['tags'])   # This function takes a purge_tags argument so feel free to tack that on; it defaults to True but is nice for users to have an option to turn it off

    if tags_to_add:
        # tag resource with ansible_dict_to_boto3_tag_list(tags_to_add)

    if tags_to_remove:
        # untag resource with tags_to_remove, no modification to tags_to_remove necessary as it's the list of keys to remove
```